detection-rules/attachment_eml_filelink_nlu.yml

```
@@ -0,0 +1,23 @@
+name: "Attachment: EML with file sharing link & suspicious language"
+description: |
+  Attached EML links to a file sharing link that contains suspicious language.
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  // exclude bounce backs & read receipts
+  and not any(attachments, .content_type == "message/delivery-status")
+  // if the "References" is in the body of the message, it's probably a bounce
+  and not any(headers.references,
+      strings.contains(body.html.display_text, .)
+  )
+  and any(attachments, .content_type == "message/rfc822"
+      and any(file.explode(.),
+          any(.scan.url.urls, .domain.domain in $free_file_hosts)
```

```suggestion
          any(.scan.url.urls, .domain.domain in $free_file_hosts or .domain.root_domain in $free_subdomain_hosts)
```

looking at the original EML, i think this might be what we want (in order to pick up the Sharepoint link)

---

Sharepoint appears in both lists, but I see your point, `.domain.domain` wouldn't catch it...

---

why not `.domain.root_domain in $free_file_hosts` as well? this is an interesting case, because of:
- a `sharepoint.com` link not matching as a `.domain.domain` but as a `.domain.root_domain`
- `drive.google.com` matching as `.domain.domain` but not as a `.domain.root_domain`

---

I'm thinking
```
any(.scan.url.urls, .domain.domain in $free_file_hosts or .domain.root_domain in $free_file_hosts)
```
or, for overkill
```
any(.scan.url.urls, .domain.domain in $free_file_hosts or .domain.root_domain in $free_file_hosts or .domain.domain in $free_subdomain_hosts or .domain.root_domain in $free_subdomain_hosts)
```

---

i wonder if sharepoint shouldn't be in $free_file_hosts... can you share a file on the root domain?

if yes then yeah we should keep both so:
```
any(.scan.url.urls, .domain.domain in $free_file_hosts or .domain.root_domain in $free_file_hosts or .domain.root_domain in $free_subdomain_hosts)
```

---

I don't think so, I've always seen `yourorg.sharepoint.com`.
so yeah, in that case, your initial suggestion makes sense to me